File: Swift/student-attendance-record-i.swift
class Solution {
    func checkRecord(_ s: String) -> Bool {
        var recordACount = 0
        var chars = [Character](s)

        for (i, record) in chars.enumerated() {
            if record == "A" {
                recordACount += 1
                if recordACount == 2 {
                    return false
                }
            }
            if record == "L" && i < chars.count - 2 && chars[i+1] == record && chars[i+2] == record {
                return false
            }
        }

        return true
    }
}

File: Python/student-attendance-record-i.py
class Solution:
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        countA = 0

        for i in range(len(s)):
            if s[i] == 'A':
                countA += 1
                if countA == 2:
                    return False
            if i < len(s) - 2 and s[i] == s[i+1] == s[i+2] == 'L':
                return False

        return True
